File: src/main/java/testbed/demo/tests/RaycastExplosionTest.java
package testbed.demo.tests;

import library.dynamics.World;
import library.explosions.RaycastExplosion;
import library.math.Vec2;
import testbed.demo.TestBedWindow;

public class RaycastExplosionTest {
    public static final String[] text = {"Raycast Explosions:", "Left click: casts an explosion"};
    public static boolean active = false;
    public static RaycastExplosion r;

    public static void load(TestBedWindow testBedWindow) {
        testBedWindow.setWorld(new World(new Vec2(0, -9.81)));
        testBedWindow.setCamera(new Vec2(0, 300), 2.0);
        World temp = testBedWindow.getWorld();
        active = true;

        testBedWindow.buildExplosionDemo();

        r = new RaycastExplosion(new Vec2(0, 1), 100, 1000, temp.getBodies());
        testBedWindow.add(r);

    }
}

File: src/test/java/demo/tests/RaycastExplosionTest.kt
package demo.tests

import demo.window.TestBedWindow
import library.dynamics.World
import library.explosions.RaycastExplosion
import library.math.Vec2

object RaycastExplosionTest {
    @JvmField
    val text = arrayOf("Raycast Explosions:", "Left click: casts an explosion")
    @JvmField
    var active = false
    @JvmField
    var r: RaycastExplosion? = null
    @JvmStatic
    fun load(testBedWindow: TestBedWindow) {
        testBedWindow.world = World(Vec2(.0, -9.81))
        testBedWindow.setCamera(Vec2(.0, 300.0), 2.0)
        val temp = testBedWindow.world
        active = true
        testBedWindow.buildExplosionDemo()
        r = RaycastExplosion(Vec2(.0, 1.0), 100, 1000, temp.bodies)
        testBedWindow.add(r)
    }
}